src/context/WalletProvider/NativeWallet/components/NativeImportKeystore.tsx

```diff
@@ -0,0 +1,229 @@
+    (e: React.DragEvent) => {
+      e.preventDefault()
+      e.stopPropagation()
+      setIsDragging(false)
+
+      const files = e.dataTransfer.files
+      if (files?.[0]) {
+        processFile(files[0])
+      }
+    },
+    [processFile],
+  )
+
+  const handleFileInput = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => {
+      const files = e.target.files
+      if (files?.[0]) {
+        processFile(files[0])
+      }
+    },
+    [processFile],
+  )
+
+  return (
+    <FormControl>
+      <Input
+        type='file'
+        accept='.txt'
+        onChange={handleFileInput}
+        id='file-upload'
+        display='none'
+        data-test='wallet-native-keystore-upload'
+      />
+      <Box
+        as='label'
+        htmlFor='file-upload'
+        w='full'
+        h='32'
+        border='2px'
+        borderStyle='dashed'
+        borderColor={isDragging ? 'blue.500' : borderColor}
+        borderRadius='xl'
+        display='flex'
+        flexDirection='column'
+        alignItems='center'
+        justifyContent='center'
+        bg={bgColor}
+        cursor='pointer'
+        transition='all 0.2s'
+        _hover={hoverSx}
+        onDragEnter={handleDragEnter}
+        onDragOver={handleDragOver}
+        onDragLeave={handleDragLeave}
+        onDrop={handleDrop}
+      >
+        <Icon as={FaFile} boxSize={6} color='gray.500' mb={2} />
+        {filename ? (
+          <CText color='gray.500'>{filename}</CText>
+        ) : (
+          <Text color='gray.500' translation='walletProvider.shapeShift.import.dragAndDrop' />
+        )}
+      </Box>
+    </FormControl>
+  )
+}
+
+export const NativeImportKeystore = ({ history }: RouteComponentProps) => {
+  const [keystoreFile, setKeystoreFile] = useState<string | null>(null)
+  const mixpanel = getMixPanel()
+
+  const translate = useTranslate()
+
+  const {
+    setError,
+    handleSubmit,
+    formState: { errors, isSubmitting },
+    register,
+  } = useForm<NativeWalletValues>({ shouldUnregister: true })
+
+  const onSubmit = useCallback(
+    async (values: FieldValues) => {
+      try {
+        const { Vault } = await import('@shapeshiftoss/hdwallet-native-vault')
+        const vault = await Vault.create()
+        vault.meta.set('createdAt', Date.now())
+
+        if (!keystoreFile) {
+          throw new Error('No keystore uploaded')
+        }
+
+        try {
+          await vault.loadFromKeystore(keystoreFile, values.keystorePassword)
+        } catch (e) {
+          setError('keystorePassword', {
+            type: 'manual',
+            message: translate('walletProvider.shapeShift.import.invalidKeystorePassword'),
+          })
+          return
+        }
+
+        history.push(NativeWalletRoutes.Password, { vault })
+        mixpanel?.track(MixPanelEvent.NativeImportKeystore)
+      } catch (e) {
+        setError('mnemonic', {
+          type: 'manual',
+          message: 'walletProvider.shapeShift.import.invalidKeystore',
+        })
```

Comment:
Probably fine to ditch this one as we never end up there - but prefer to keep it in case we manage to detect the imported file is a keystore or not, in which case we could move this catch and reach this error-handling

Reply:
As mentioned above, we never end up here in the first place, it's more of a futureproofing kind-of-thing if we ever manage to handle that... but on the second read of this meh, dead code is useless.

39850e160